apps/ui/src/components/inputs/editTextInput.tsx

```diff
@@ -0,0 +1,189 @@
+import type { Metadata } from '@bako-id/sdk';
+import {
+  Box,
+  Flex,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+  type InputProps,
+} from '@chakra-ui/react';
+import { useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { Dialog } from '../dialog';
+import { CloseIcon } from '../icons/closeIcon';
+import { TrashIcon } from '../icons/trashIcon';
+
+interface IEditTextValueInput extends InputProps {
+  title: string;
+  modalType: string;
+  onSave: (metadata: Metadata) => Promise<void>;
```

Comment:
It's not advisable to use a promise here in this component. The input should be completely synchronous and decoupled from any asynchronous rule/call. Its only responsibility is to set the value in the originating state and perform the necessary validations.